src/app/modules/courses/components/search/search.component.ts

```diff
@@ -1,18 +1,28 @@
-import { Component, EventEmitter, Output } from '@angular/core';
+import { Component, EventEmitter, OnInit, Output } from '@angular/core';
+import { debounceTime, filter, Subject } from 'rxjs';
 
 @Component({
   selector: 'app-search',
   templateUrl: './search.component.html',
   styleUrls: ['./search.component.scss']
 })
-export class SearchComponent {
+export class SearchComponent implements OnInit{
 
   @Output() search = new EventEmitter<string>();
 
+  subject$ = new Subject<string>();
+
   searchText: string;
 
+  ngOnInit(): void {
+    this.subject$.pipe(
+      filter(res => res?.length >= 3),
```

Comment:
What happens, when user will enter some word (more then 3 chars) and then decided to remove all and see original courses (which was before search). Will he able to do it?

Reply:
there were a task to add this logic, no idea why

Reply:
yes and it is correct, but you shouldn't remove original logic

Reply:
I would say the better solution here should be: res => res?.length >= 3 || !res?.length